accepted/2021/winforms/streamline-application-bootstrap.md

```diff
@@ -0,0 +1,302 @@
+
+### Stretch Goals
+
+1. Update Visual Studio property page for Windows Forms projects.
+2. Build the following Roslyn Analyzers functions:
+
+    * Check for invocations of now-redundant `Applicaiton.*` methods invoked by `ApplicationConfiguration.Initialize()`.
+    * If a custom app.manifest is specified, parse it, and if dpi-related settings are found - warn the user, and direct to supply the dpi mode via the MSBuild property defined below.
+    * (Consider) checking for app.config and dpi-related configurations, if found - warn the user, and direct to supply the dpi mode via the MSBuild property defined below.
+    * Check if `Application.SetHighDpiMode()` is invoked with anything other than `HighDpiMode.PerMonitorV2` (see: [dotnet/winforms-designer#3278](https://github.com/dotnet/winforms-designer/issues/3278))
+
+
+### Non-Goals
+
+<!--
+Provide a bullet point list of aspects that your feature does not need to do.
+The goal of this section is to cover problems that people might think you're
+trying to solve but deliberately would like to scope out. You'll likely add
+bullets to this section based on early feedback and reviews where requirements
+are brought that you need to scope out.
+-->
+
+1. Modify the Windows Forms Designer process to read values either from *.myapp file (VB) or MSBuild properties (C#). This work will be tracked in separately, targeting Dev17.
+1. Migrate Visual Basic apps off the Application Framework or change the use of *.myapp file.
+1. Streamline the bootstrap of Visual Basic apps or use Roslyn source generators for this purpose.
+1. Use Roslyn analyzers in Visual Basic scenarios until previous two items are addressed.
+
+
+:thinking: Strategically it could be benefitial to migrate Visual Basic off the Application Framework and *.myapp file in favour of Roslyn source generators and MSBuild properties. This could also significantly reduce efforts in maintaining Visual Studio property pages for Visual Studio projects (e.g. [dotnet/project-system#7236](https://github.com/dotnet/project-system/issues/7236), [dotnet/project-system#7240](https://github.com/dotnet/project-system/issues/7240), and [dotnet/project-system#7241](https://github.com/dotnet/project-system/issues/7241))
+
+
+
+## Stakeholders and Reviewers
+
+<!--
+We noticed that even in the cases where we have specs, we sometimes surprise key
+stakeholders because we didn't pro-actively involve them in the initial reviews
+and early design process.
+
+Please take a moment and add a bullet point list of teams and individuals you
+think should be involved in the design process and ensure they are involved
+(which might mean being tagged on GitHub issues, invited to meetings, or sent
+early drafts).
+-->
+
+* Windows Forms runtime/designer: @dotnet/dotnet-winforms @OliaG @DustinCampbell 
+* Visual Basic: @KathleenDollard  @KlausLoeffelmann
+* Roslyn: @jaredpar @sharwell
+* Project System: @drewnoakes 
+* SDK: @ericstj
```

Comment:
I work on libraries not SDK. It would be good to involve @dsplaisted for SDK

Reply:
Apologies. Corrected.
I tagged you primarily because of your proposal https://github.com/dotnet/designs/pull/181